### PR TITLE
Remove Layer requirement of PointwiseMultiplicative and ElementaryFunctions

### PR DESCRIPTION
**Pull request by Shashi456**
Fix: #343 

I was wondering if i should add a FIXIT by mentioning we reverted the change.

### CODE REVIEW

**Comment by rxwei** [comment]
I don't think it would be this trivial. Did you verify that it compiles?

**Comment by Shashi456** [comment]
Yeah, forgot to do it it on the latest build.  just pulling it and compiling right now.

File: Sources/TensorFlow/Layers/Core.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
 /// Dropout consists in randomly setting a fraction of input units to `0` at each update during
 /// training time, which helps prevent overfitting.
 @frozen
-public struct Dropout<Scalar: TensorFlowFloatingPoint>: ParameterlessLayer {
+public struct Dropout<Scalar: TensorFlowFloatingPoint>: Layer {
```
Comment (rxwei) [inline Sources/TensorFlow/Layers/Core.swift]: These should _not_ be changed.

File: Sources/TensorFlow/Layer.swift [hunk 2]
```diff
@@ -44,8 +43,7 @@ public extension Layer {
 }
 
 /// An empty struct representing empty `TangentVector`s for parameterless layers.
-public struct EmptyTangentVector: Differentiable, VectorProtocol, ElementaryFunctions,
-                                  PointwiseMultiplicative, KeyPathIterable {
+public struct EmptyTangentVector: Differentiable, VectorProtocol, KeyPathIterable {
```
Comment (rxwei) [inline Sources/TensorFlow/Layer.swift]: Conformances should not be dropped from this type. Only the `Layer` protocol.

**Comment by Shashi456** [comment]
@rxwei yeah I will revert them, I pushed them to this branch by mistake. I was just getting some SIL errors and changed the wrong conformance.

**Comment by Shashi456** [comment]
@rxwei You were right it's not as trivial. One evident error is ` error: type 'Function<Input, Output>.AllDifferentiableVariables' does not conform to protocol 'VectorProtocol'`

**Comment by Shashi456** [comment]
@rxwei this commit passes on the latest build without any errors.

**Comment by eaplatanios** [comment]
This should make layers not directly usable with the momentum-based optimizers such as Adam, AdaGrad, RMSProp, etc. I would be surprised if it doesn't. Maybe we can manually add them to layers that can conform to them so we keep them compatible with the optimizers.

**Comment by eaplatanios** [comment]
It also breaks SGD. I just tested on my machine.

**Comment by Shashi456** [comment]
@eaplatanios I was wondering that it would break optimizers. 

@rxwei closed #343. I'll close this too.

**Pull request closed without merging**